squire_lib/src/ffi_tournament.rs

```diff
@@ -280,21 +337,25 @@ pub extern "C" fn load_tournament_from_file(__file: *const c_char) -> Tournament
         }
     };
 
-    // Cannot open the same tournament twice
-    if FFI_TOURNAMENT_REGISTRY
-        .get()
-        .unwrap()
-        .contains_key(&tournament.id)
-    {
-        println!("Input file is already open");
-        return TournamentId::new(Uuid::from_bytes(NULL_UUID_BYTES));
+    unsafe {
+        // Cannot open the same tournament twice
+        if FFI_TOURNAMENT_REGISTRY
+            .get_mut()
+            .unwrap()
+            .contains_key(&tournament.id)
+        {
+            println!("[FFI]: Input tournament is already open");
+            return TournamentId::new(Uuid::from_bytes(NULL_UUID_BYTES));
+        }
     }
 
     let tid: TournamentId = tournament.id.clone();
-    FFI_TOURNAMENT_REGISTRY
-        .get()
-        .unwrap()
-        .insert(tid, tournament.clone());
+    unsafe {
+        FFI_TOURNAMENT_REGISTRY
+            .get_mut()
+            .unwrap()
+            .insert(tid, tournament.clone());
```

Comment:
Unnecessary `clone`. you can just return `tid`. This is also in an unnecessary `unsafe` block.

Reply:
Resolved in acaed8005beb87ce9eb8914db6a871079f558b76